test/e2e-common.sh

```diff
@@ -0,0 +1,83 @@
+#!/usr/bin/env bash
+
+# Copyright 2019 The Knative Authors
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+
+# This script runs the end-to-end tests for the kn client.
+
+# If you already have the `KO_DOCKER_REPO` environment variable set and a
+# cluster setup and currently selected in your kubeconfig, call the script
+# with the `--run-tests` argument and it will use the cluster and run the tests.
+
+# Calling this script without arguments will create a new cluster in
+# project $PROJECT_ID, start Knative serving, run the tests and delete
+# the cluster.
+
+source $(dirname $0)/../vendor/github.com/knative/test-infra/scripts/e2e-tests.sh
+
+DEFAULT_BUILD=latest
+
+# This is the variable saving the link to the YAML file to create and delete knative serving.
+KNATIVE_SERVING_RELEASE_YAML=""
+
+# Install the stable release Knative/serving in the current cluster.
+# This method is copied from github.com/knative/test-infra/scripts/e2e-tests.sh, since the original var
+# $KNATIVE_SERVING_RELEASE is readonly. In order to make it changeable, we rewrite the method with another
+# variable $KNATIVE_SERVING_RELEASE_YAML.
+function start_release_knative_serving() {
```

Comment:
I see your motivation but we shouldn't do this, because:

* We don't benefit from changes in this method upstream
* When the overridden function gets renamed the script fails silently (because this method is just not called).

Instead we should push a change upstream to make the version to install parameterizable either via an argument or an environment variable.

Tbh I think its even a fault that this method _can_ be overridden. This subtle coupling is really hard to debug.

Let me open an issue in the testing upstream repo.

Reply:
Yep, it is good to see test-infra is able to support the installation of custom release number.